### PR TITLE
Render specific gists based on ID

### PR DESCRIPTION
**Pull request by captainsafia**
Doesn't exactly work yet.... :/

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
+const Main = React.createClass({
+  render() {
+    return <h1>
+      Loading Notebook...
+    </h1>
+  }
+});
 
-const gistID = gistIDs[Math.floor(Math.random() * gistIDs.length)];
+const Notebook = React.createClass({
```
Comment (rgbkrk) [inline src/index.js:30]: It's ok if you want to use JSX and ES6 classes here, as there is babel setup for it.

File: src/index.js [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
-    }
-  })
-  .then((nbJSON) => {
-    render(<NotebookPreview notebook={nbJSON}/>,
-           document.getElementById('root'));
-  })
+render((
+  <Router history={browserHistory}>
+    <Route path="/" component={Notebook}>
+      <Route path="/gist/:gistId" component={Notebook}/>
```
Comment (rgbkrk) [inline src/index.js]: 👍 

Should we mirror the path structure of nbviewer?

File: src/index.js [hunk 3]
```diff
@@ -1,45 +1,53 @@
...
-        }
+              const nbString = fileResponse.content;
+              const nbJSON = JSON.parse(nbString);
+              return nbJSON;
+            }
+          }
+        })
+        .then((nbJSON) => {
+          return nbJSON;
+        })
```
Comment (rgbkrk) [inline src/index.js]: We should definitely move these out into a separate function. It should make it easier to test and iterate.

On top of that, we could certainly handle state from the top down (flux/redux style). It's probably not too big a deal though.

**Comment by captainsafia** [comment]
I originally had more routes on this than is present but I'm settling on the following. If you hit `/` you get redirected to a random gist and if you hit `gist/:gistId` the gist with `gistId` is rendered.

Everything builds fine, but it seems that the `IndexRedirect` isn't working, namely it doesn't redirect to `gist/randomId` and trying to render something on `gist/gistId` doesn't render anything.

To clarify things, the URLs I'm hitting on my local machine are.
- `file:///Users/captainsafia/dev/notebook-preview-demo/index.html#/`
- `file:///Users/captainsafia/dev/notebook-preview-demo/index.html#/gist/21ab98d355a7fc681883`

I tried to debug the issue with the `gist/:gistId` component not rendering but most documentation on issues with components not rendering are related to nested routes, which this is not. It looks like the `render` function isn't being call when that route is hit.

I also tried this with nested routes to no avail. I think the route of the problem might be the way that I've structured the routes in the top-level in `render`.

**Comment by captainsafia** [comment]
OK! This works now. There are still a couple of issues. Namely, how `react-router`'s `browserHistory` handles page refresh.

**Comment by rgbkrk** [comment]
Awesome. What are the difficulties being run into with respect to react router and history.

**Pull request merged**